client-js/client/client.ts

```diff
@@ -0,0 +1,707 @@
+  }
+
+  public enableScreenShare(enable: boolean) {
+    return this._transport.enableScreenShare(enable);
+  }
+
+  public get isSharingScreen(): boolean {
+    return this._transport.isSharingScreen;
+  }
+
+  // ------ Messages
+
+  /**
+   * Directly send a message to the bot via the transport.
+   * Do not await a response.
+   * @param msgType - a string representing the message type
+   * @param data - a dictionary of data to send with the message
+   */
+  @transportReady
+  public sendClientMessage(msgType: string, data?: unknown): void {
+    this._transport.sendMessage(
+      new RTVIMessage(RTVIMessageType.CLIENT_MESSAGE, {
+        t: msgType,
+        d: data,
+      } as ClientMessageData)
+    );
+  }
+
+  /**
+   * Directly send a message to the bot via the transport.
+   * Wait for and return the response.
+   * @param msgType - a string representing the message type
+   * @param data - a dictionary of data to send with the message
+   * @param timeout - optional timeout in milliseconds for the response
+   */
+  @transportReady
+  public async sendClientRequest(
+    msgType: string,
+    data: unknown,
+    timeout?: number
+  ): Promise<unknown> {
+    return new Promise((resolve, reject) => {
+      const msgData: ClientMessageData = { t: msgType, d: data };
+      this._messageDispatcher
+        .dispatch(msgData, RTVIMessageType.CLIENT_MESSAGE, timeout)
+        .then((response) => {
+          logger.debug("[RTVI Client] Response received", response);
+          const data = response.data as ClientMessageData;
+          resolve(data.d);
+        })
+        .catch((e) => {
+          reject(e);
+        });
+    });
+  }
+
+  public registerFunctionCallHandler(
+    functionName: string,
+    callback: FunctionCallCallback
+  ) {
+    this._functionCallCallbacks[functionName] = callback;
+  }
+
+  @transportReady
+  public async appendToContext(context: LLMContextMessage) {
+    const response = await this._messageDispatcher.dispatch(
+      {
+        role: context.role,
+        content: context.content,
+        run_immediately: context.run_immediately,
+      } as LLMContextMessage,
+      RTVIMessageType.APPEND_TO_CONTEXT
+    );
+    const responseData = response.data as AppendToContextResultData;
+    return !!responseData.result;
+  }
+
+  /**
+   * Disconnects the bot, but keeps the session alive
+   */
+  @transportReady
+  public disconnectBot(): void {
+    this._transport.sendMessage(
+      new RTVIMessage(RTVIMessageType.DISCONNECT_BOT, {})
+    );
+  }
+
+  protected handleMessage(ev: RTVIMessage): void {
+    logger.debug("[RTVI Message]", ev);
+
+    switch (ev.type) {
+      case RTVIMessageType.BOT_READY: {
+        const data = ev.data as BotReadyData;
+        const botVersion = data.version.split(".").map(Number);
+        if (botVersion[0] < 1) {
+          logger.warn(
+            "[Pipecat Client] Bot version is less than 1.0.0, which may not be compatible with this client."
```

Comment:
So the client is only going to check whether the server (the bot) is too old, not whether it's too new for the client? (This is a future-facing concern, and a question about what our general versioning approach will be)